pkg/controller/localvolume/add.go

```diff
@@ -0,0 +1,129 @@
+package localvolume
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+
+	secv1client "github.com/openshift/client-go/security/clientset/versioned/typed/security/v1"
+	localv1 "github.com/openshift/local-storage-operator/pkg/apis/local/v1"
+	"github.com/openshift/local-storage-operator/pkg/common"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	componentName       = "local-storage-operator"
+	localDiskLocation   = "/mnt/local-storage"
+	ownerNamespaceLabel = "local.storage.openshift.io/owner-namespace"
+	ownerNameLabel      = "local.storage.openshift.io/owner-name"
+
+	localVolumeFinalizer = "storage.openshift.com/local-volume-protection"
+)
+
+// ReconcileLocalVolume reconciles a LocalVolume object
+type ReconcileLocalVolume struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client            client.Client
+	secClient         secv1client.SecurityV1Interface
+	scheme            *runtime.Scheme
+	apiClient         apiUpdater
+	lvMap             *common.StorageClassOwnerMap
+	controllerVersion string
+
+	localStorageNameSpace string
+	localDiskLocation     string
+	provisonerConfigName  string
+	diskMakerConfigName   string
```

Comment:
secClient, scheme, controllerVersion, localStorageNameSpace, localDiskLocation, provisonerConfigName and diskMakerConfigName are not used (or only written to).

Reply:
Removed